src/Microsoft.Data/StringColumn.cs

```diff
@@ -0,0 +1,225 @@
+
+        public StringColumn(string name, long length) : base(name, length, typeof(string))
+        {
+            _stringBuffers.Add(new List<string>());
+        }
+
+        public StringColumn(string name, IEnumerable<string> values) : base(name, 0, typeof(string))
+        {
+            values = values ?? throw new ArgumentNullException(nameof(values));
+            if (_stringBuffers.Count == 0)
+            {
+                _stringBuffers.Add(new List<string>());
+            }
+            foreach (var value in values)
+            {
+                Append(value);
+            }
+        }
+
+        public void Append(string value)
+        {
+            List<string> lastBuffer = _stringBuffers[_stringBuffers.Count - 1];
+            if (lastBuffer.Count == int.MaxValue)
+            {
+                lastBuffer = new List<string>();
+                _stringBuffers.Add(lastBuffer);
+            }
+            lastBuffer.Add(value);
+            Length++;
+        }
+        private int GetBufferIndexContainingRowIndex(ref long rowIndex)
+        {
+            if (rowIndex > Length)
+            {
+                throw new ArgumentOutOfRangeException(Strings.ColumnIndexOutOfRange, nameof(rowIndex));
+            }
+            int curArrayIndex = 0;
+            int numBuffers = _stringBuffers.Count;
+            while (curArrayIndex < numBuffers && rowIndex > _stringBuffers[curArrayIndex].Count)
+            {
+                rowIndex -= _stringBuffers[curArrayIndex].Count;
+                curArrayIndex++;
+            }
+            return curArrayIndex;
+        }
+
+        protected override object GetValue(long rowIndex)
+        {
+            int bufferIndex = GetBufferIndexContainingRowIndex(ref rowIndex);
+            return _stringBuffers[bufferIndex][(int)rowIndex];
+        }
+
+        protected override object GetValue(long startIndex, int length)
+        {
+            var ret = new List<string>();
+            int bufferIndex = GetBufferIndexContainingRowIndex(ref startIndex);
+            while (ret.Count < length && bufferIndex < _stringBuffers.Count)
+            {
+                for (int i = (int)startIndex; ret.Count < length && i < _stringBuffers[bufferIndex].Count; i++)
+                {
+                    ret.Add(_stringBuffers[bufferIndex][i]);
+                }
+                bufferIndex++;
+                startIndex = 0;
+            }
+            return ret;
+        }
+
+        protected override void SetValue(long rowIndex, object value)
+        {
+            if (value is string)
+            {
+                int bufferIndex = GetBufferIndexContainingRowIndex(ref rowIndex);
+                _stringBuffers[bufferIndex][(int)rowIndex] = (string)value;
+            }
+            else
+            {
+                throw new ArgumentException(Strings.MismatchedValueType + " string", nameof(value));
```

Comment:
When throwing exceptions with a  message that comes from a resource, you shouldn't concat strings. Instead use formatted strings - `{0}`. The reason is because when translating to other languages, the hard-coded type name may need to move in the sentence to make sense in the language.